### ISSUE TITLE
InvalidResourceId Error on Windows for SSM Parameters with Slash in Name

### ISSUE THREAD

**Issue by cm-1** [aws/aws-cli#4419]
Hi there. I was attempting to use the aws cli to manage tags on some SSM Parameters I had created; specifically, I was using the list-tags-for-resource command. For most parameters, this works just fine. However, I have a few parameters with a "/" in their name, e.g. "/build-number/ExcelService". When I try to try to query the tags for these parameters, I receive an error.

Command:
```
aws ssm list-tags-for-resource --resource-type "Parameter" --resource-id "/build-number/ExcelService"
```

Error:
```
An error occurred (InvalidResourceId) when calling the ListTagsForResource operation:
```

(Note: the trailing colon in the error message is not a copy-paste mistake; nothing followed it in the error message).

The same error is also thrown for the command add-tags-to-resource. I've tried escaping the slashes with backslashes, tried removing the leading slash, and various other modifications, but nothing avoids the error.

Additionally, after posting the issue on the AWS forums ([see here](https://forums.aws.amazon.com/thread.jspa?messageID=912582)), it would appear that this issue is Windows-specific, regardless of whether it was installed using the .msi or using pip.

**Comment by kyleknap** [comment]
I was not able to reproduce either. I tried it on Windows (using PowerShell) and did not have any issues. Could you provide debug logs from running the command? You can do so by append `--debug` to the command you ran. Also out of curiosity, are you using the correct region when running the command? That could be one reason why you are getting the resource not found error back.

**Comment by cm-1** [comment]
Thanks for the reply. I probably won't be able to reproduce it by the time you want to close this issue, unfortunately. This happened during summer work for a company whose AWS account I no longer have access to (since I'm back in school now), and I want to prepare a bit before getting my own AWS account to take full advantage of the 12-month free tier. I could maybe reach out to a co-worker and get them to run this for me, though.

If it helps (and sorry for not mentioning it in the first post), I encountered the issue when using Git Bash. I believe I also tested either Command Prompt or Powershell and still saw the issue, but I can't remember for certain, so perhaps that's the reason for the difference. I know definitively that the region was not the issue, since I tested resources without "/" in their names in the same region, without changing the region in the command, and encountered no issue for them.
